_static/link_gen/link.js

```diff
@@ -141,28 +154,58 @@ function changeTab(div) {
  * See https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L276
  */
 function generateCloneDirectoryName(gitCloneUrl) {
+    if(gitCloneUrl.slice(-1) == "/")
+        gitCloneUrl = gitCloneUrl.slice(0,-1);
     var lastPart = gitCloneUrl.split('/').slice(-1)[0];
     return lastPart.split(':').slice(-1)[0].replace(/(\.git|\.bundle)?/, '');
 }
```

Comment:
I don't understand this change, can you help me by describing what you are doing here?

Preferably, this function would already have a documentation string to clarify its purpose and typical input it can expect and should be able to handle. I assume you are making this function support a new kind of input by this change - is that correct?

Reply:
I was responding to what happens if you add a forward slash to the end of the git clone URL. The function tries to split by forward slash and you end up without the name you wanted. I watched it happen when we were running training sessions for Community College users. Probably should have done the separate commit for the change or not included it here. Pull it out?

Reply:
If this can be a 2-line change, and its unrelated to tests etc, then yeah ok pull it out - i dont want you to need to invest much time in it further otherwise though.

I think overall, reading splice/split/index etc is hard to read, so having a comment clarifying what goes on can be helpful.

Reply:
ok I left it in and added some documentation